docs/manual/developer/04_style_guide.md

```diff
@@ -0,0 +1,177 @@
+  * If defined must have at least one of the following sub-sections:
+    * `general`
+    * `dependency`
+    * `performance`
+    * `management`
+    * `functionality`
+* `conflicts`
+  * Must be a valid rule id
+* `requires`
+  * * Must be a valid rule id
+* `template`
+
+### Group
+This section describes the styleguide around the `group.yml` files. 
+All the above [YAML](manual/developer/04_style_guide:yaml) rules apply.
+A group should only contain rules that effect the same software or service.
+
+#### Sections
+Group sections should be in the following order, if present.
+* `documentation_complete`
+* `title`
+  * Must be in [Title case](https://en.wikipedia.org/wiki/Title_case)
+* `description` (HTML-Like)
+
+### Benchmark
+This section describes the styleguide around the `benchmark.yml` files. 
+All the above [YAML](manual/developer/04_style_guide:yaml) rules apply.
+
+#### Sections
+Benchmark sections should be in the following order, if present.
+* `documentation_complete`
+* `status`
+* `title`
+  * Must be in [Title case](https://en.wikipedia.org/wiki/Title_case)
+* `description` (HTML-Like)
+* `notice` (HTML-Like)
+* `front-matter` (HTML-Like)
+* `rear-matter` (HTML-Like)
+* `version`
+
+## Remediation
+All remediations should have the following header with the correct 
+```bash
+# platform = multi_platform_all
+# reboot = false
+# strategy = enable
+# complexity = low
+# disruption = low
+```
+
+### Bash
+* Should use marcos instead shared functions
+* Use 4-space indentation
+* Put `do` or `then` on the same line as `for` or `if` respectively, e.g. `for file in *; do`
+
+### Ansible
+* Shall follow all the rules in the [YAML](manual/developer/04_style_guide:yaml) section
+* Should prefer using Ansible modules over just calling system commands
+
+
+## XML
+* Shall use the `.xml` for the file extension
+* Must be able to be parsed by Python's XML parser
+
+### OVAL
+* The `id` attribute of `<definition>` should be `{{{ rule_id }}}`
+* The elements should be in the following order:
+  * `def-group`
+    * `definition`
+      * `metadata`
+        * Should be defined by the`oval_metadata` marco
+      * `criteria`
+    * * Like OVAL `test`, `object`, and `state` should have the same name after there respective prefix.
+    * Like OVAL `test`, `object`, and `state` should group together, if there are many tests the order below should be repeated for each group.
+      * `test`
+      * `object`
+      * `state`
+
+### Test elements
+* `id` should start with `test_`
+
+### Object elements
+* `id` should start with `obj_`
+
+### State elements
+* `id` should start with `state_`
+
+## Tests
+* Should always do something, even if testing default behavior
+* Should test one change
+
+## Markdown
+* [One sentence per line](https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line)
+
+### Headings
+* Must have one blank file above headings
+* Must be in [Title case](https://en.wikipedia.org/wiki/Title_case)
```

Comment:
Should this apply to all heading levels?
I see, for example,  in line 165 `### State elements`, should it be `### State Elements`?